odbc_fdw.c

```diff
@@ -174,7 +177,62 @@ static struct odbcFdwOption valid_options[] =
 	{ NULL,       InvalidOid}
 };
 
-typedef enum { TEXT_CONVERSION, HEX_CONVERSION, BIN_CONVERSION, BOOL_CONVERSION } ColumnConversion;
+typedef enum { TEXT_CONVERSION, BIN_CONVERSION, BOOL_CONVERSION } ColumnConversion;
+
+static GetDataTruncation
+result_truncation(SQLRETURN ret, SQLHSTMT stmt)
+{
+    SQLCHAR sqlstate[5];
+    GetDataTruncation truncation = NO_TRUNCATION;
+    if (ret == SQL_SUCCESS_WITH_INFO)
+    {
+		SQLGetDiagRec(SQL_HANDLE_STMT, stmt, 1, sqlstate, NULL, NULL, 0, NULL);
+        if (strcmp((char*)sqlstate, ODBC_SQLSTATE_STRING_TRUNCATION) == 0)
+        {
+            truncation = STRING_TRUNCATION;
+        }
+		else if (strcmp((char*)sqlstate, ODBC_SQLSTATE_FRACTIONAL_TRUNCATION) == 0)
+		{
+            truncation = FRACTIONAL_TRUNCATION;
+        }
+    }
+    return truncation;
+}
+
+static void
+resize_buffer(char ** buffer, int *size, int used_size, int required_size)
+{
+    if (required_size > *size)
```

Comment:
Why not use repalloc? Am I missing something?

Reply:
The whole function brings a lot of complexity that I'm not sure should be here at all, like only freeing the buffer if used_size because it's relying on buffer being NULL (not allocated).

Reply:
No, it was me who was missing the existence of `repalloc`. I'll review this later if I have time